tests/ParserTest.elm

```diff
@@ -605,6 +605,85 @@ expr =
                   , "\"'\""
                   , Ok (Literal (String "'"))
                   )
+                , ( "double quote"
+                  , "\"\\\"\""
+                  , Ok (Literal (String "\""))
+                  )
+                , ( "escape n"
+                  , "\"\\n\""
+                  , Ok (Literal (String "\n"))
+                  )
+                , ( "escape r"
+                  , "\"\\r\""
+                  , Ok (Literal (String "\u{000D}"))
+                  )
+                , ( "escape t"
+                  , "\"\\t\""
+                  , Ok (Literal (String "\t"))
+                  )
+                , ( "emoji"
+                  , "\"😃\""
+                  , Ok (Literal (String "😃"))
+                  )
+                , ( "escaped unicode code point"
+                  , "\"\\u{1F648}\""
+                  , Ok (Literal (String "🙈"))
+                  )
+                , ( "combo of escapes and chars"
+                  , "\"\\u{1F648}\\n\\r\\t\\abc123\""
+                  , Ok (Literal (String "🙈\n\r\tabc123"))
```

Comment:
Is this `\r` consistent with the other test - that returns `\u{000D}`? If both would pass, why write the `\u{000D}` thing at all?

Reply:
`\r` and `\u{000D}` are exactly the same characters.

However, elm-format prefers that we use `\u{000D}` instead of `\r`. I assume that the Elm overlords are trying to deprecate `\r` or something, but, in the meantime, I assume we still need to know how to parse `\r` even if we never write it that way in our source code.

Unless you want to use `\r` everywhere, I should:
```diff
- , Ok (Literal (String "🙈\n\r\tabc123"))
+ , Ok (Literal (String "🙈\n\u{000D}\tabc123"))
```

Reply:
OK, now that you mention it I remember `\r` being changed by `elm-format` under my hands. Let's keep it as you wrote it then.